src/types.rs

```diff
@@ -475,3 +506,209 @@ impl<'buf> RecvMsgOut<'buf> {
         self.header.flags
     }
 }
+
+/// [MatchOn] is a builder for constructing match criteria for request cancelation.
```

Comment:
Cancellation. I see that `cancelation` can be said to be technically correct, but it appears very infrequently in software. Now that I've seen it used here a few times, I've grown to like it, but still,
it's not used in the `tokio` library while the form with two els is used often.